python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -0,0 +1,284 @@
+        localFeatures, localLabels = self._getNumpyFeaturesAndLabels(dataset)
+        localFeaturesBc = sc.broadcast(localFeatures)
+        localLabelsBc = None if localLabels is None else sc.broadcast(localLabels)
+
+        # Broadcast Keras model (HDF5) file content as bytes
+        modelBytes = self._loadModelAsBytes()
+        modelBytesBc = sc.broadcast(modelBytes)
+
+        # Obtain params for this estimator instance
+        baseParamMap = self.extractParamMap()
+        baseParamDict = dict([(param.name, val) for param, val in baseParamMap.items()])
+        baseParamDictBc = sc.broadcast(baseParamDict)
+
+        # Create useful parameters from paramMap
+        def _local_fit(override_param_map):
+            """
+            Fit locally a model with a combination of this estimator's param plus
+            with additional parameters provided by the input.
+            :param override_param_map: dict, key type is MLllib Param
+                                       They are meant to override the base estimator's params.
+            :return: serialized Keras model bytes
+            """
+            # Update params
+            params = baseParamDictBc.value
+            override_param_dict = dict([
+                (param.name, val) for param, val in override_param_map.items()])
+            params.update(override_param_dict)
+
+            # Create Keras model
+            model = kmutil.bytes_to_model(modelBytesBc.value)
+            model.compile(optimizer=params['optimizer'], loss=params['loss'])
+
+            # Retrieve features and labels and fit Keras model
+            features = localFeaturesBc.value
+            labels = None if localLabelsBc is None else localLabelsBc.value
+            _fit_params = params['kerasFitParams']
+            model.fit(x=features, y=labels, **_fit_params)
+
+            return kmutil.model_to_bytes(model)
+
+        kerasModelBytesRDD = paramMapsRDD.map(lambda paramMap: (paramMap, _local_fit(paramMap)))
+        return self._collectModels(kerasModelBytesRDD)
+
+    def _loadModelAsBytes(self):
+        """
+        (usable on driver only)
+        Load the Keras model file as a byte string.
+        :return: str containing the model data
+        """
+        with open(self.getModelFile(), mode='rb') as fin:
+            fileContent = fin.read()
+        return fileContent
+
+    def _fit(self, dataset):  # pylint: disable=unused-argument
+        err_msgs = ["This function should not have been called",
+                    "Please contact library mantainers to file a bug"]
+        raise NotImplementedError('\n'.join(err_msgs))
+
+    def _validateFitParams(self, params):
+        """ Check if an input parameter set is valid """
+        if isinstance(params, (list, tuple, dict)):
+            assert self.getInputCol() not in params, \
+                "params {} cannot contain input column {}".format(params, self.getInputCol())
+
+    def fit(self, dataset, params=None):
```

Comment:
Nit: move this function before all the private ones, to put all the public pictures together.